File: Atelier4/app/src/main/java/com/example/atelier4/MainActivity.java
package com.example.atelier4;

import android.media.browse.MediaBrowser;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import androidx.media3.common.MediaItem;
import androidx.media3.exoplayer.DefaultLoadControl;
import androidx.media3.exoplayer.ExoPlayer;
import androidx.media3.exoplayer.trackselection.DefaultTrackSelector;
import androidx.media3.ui.PlayerView;

public class MainActivity extends AppCompatActivity {

    ExoPlayer player;
    PlayerView playerView;
    private final String mp3url = "https://storage.googleapis.com/uamp/The_Kyoto_Connection_-_Wake_Up/01_-_Intro_-_The_Way_Of_Waking_Up_feat_Alan_Watts.mp3";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        player = new ExoPlayer.Builder(this).build();
        playerView = findViewById(R.id.player_view);
    }

    private void initialiserPlayer(){
        playerView.setPlayer(player);

        MediaItem mediaItem = MediaItem.fromUri(mp3url);
        player.setMediaItem(mediaItem);
        player.prepare();
        player.play();
        
    }

    @Override
    protected void onStart() {
        super.onStart();
        initialiserPlayer();

    }
}

File: Annexe1B/app/src/main/java/com/example/annexe1b/MainActivity.kt
package com.example.annexe1b

import android.os.Bundle
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.nio.Buffer

class MainActivity : AppCompatActivity() {
    lateinit var nbLignes: TextView
    lateinit var nbChar: TextView
    lateinit var nbC: TextView
    lateinit var nbMots: TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        nbLignes = findViewById(R.id.textView)
        nbChar = findViewById(R.id.textView2)
        nbC = findViewById(R.id.textView3)
        nbMots = findViewById(R.id.textView4)


        nbLignes.append(lireNbLignes().toString())
        nbChar.append(lireNbChar().toString())
        nbC.append(lireNbC().toString())
        nbMots.append(ecrireText("Michael Turner").toString())
    }

    fun lireNbLignes(): Long{
        var nbLignes : Long = 0

        val fis : FileInputStream = openFileInput("blabla.txt")
        val isw = InputStreamReader(fis)
        val br = BufferedReader(isw)

        while(br.readLine() != null) nbLignes++
        br.close()
        return nbLignes
    }

    fun lireNbChar(): Int{
        var nbChar = 0

        val fis : FileInputStream = openFileInput("blabla.txt")
        val isw = InputStreamReader(fis)
        val br = BufferedReader(isw)
        br.use {
            while(br.readLine() != null){
            nbChar += br.readLine().length
        }
             }
        return nbChar
    }

    fun lireNbC(): Int{
        var nbC = 0
        val charC = 'c'
        val charCUpper = 'C'
        var lignes: String?

        val fis : FileInputStream = openFileInput("blabla.txt")
        val isw = InputStreamReader(fis)
        val br = BufferedReader(isw)

        while(true){
            lignes = br.readLine() ?: break

            if(lignes.contains(charC, ignoreCase = true)){
                nbC++
            }
        }
        br.close()
        return nbC
    }

    fun ecrireText(texte: String){
        val fis : FileOutputStream = openFileOutput("blabla.txt", MODE_APPEND)
        val isw = OutputStreamWriter(fis)
        val br = BufferedWriter(isw)

        br.newLine()
        br.write(texte)
        br.close()
    }
}